assets/app/scripts/directives/events.js

```diff
@@ -5,22 +5,106 @@ angular.module('openshiftConsole')
+            return event.involvedObject.kind === $scope.resourceKind &&
+                   event.involvedObject.name === $scope.resourceName;
+          });
+        };
+
+        var sortedEvents = [];
+        var sortEvents = function() {
+          // TODO: currentField is renamed in angular-patternfly 3.0
+          var sortID = _.get($scope, 'sortConfig.currentField.id', 'lastTimestamp'),
+              order = $scope.sortConfig.isAscending ? 'asc' : 'desc';
+          sortedEvents = _.sortByOrder($scope.events, [sortID], [order]);
+        };
+
+        var applyFilter = $filter('filter');
+        var filterForKeyword = function() {
+          $scope.filteredEvents = applyFilter(sortedEvents, $scope.filter.text);
+        };
+
+        $scope.$watch('filter.text', _.debounce(function() {
+          $scope.$apply(filterForKeyword);
+        }), 250, { maxWait: 500 });
 
-        var filterEvent = function(event) {
-          return (event.involvedObject.kind === $scope.resourceKind) && (event.involvedObject.name === $scope.resourceName);
+        var update = function() {
+          // Sort first so we can update the filter as the user types without resorting.
+          sortEvents();
+          filterForKeyword();
         };
 
+        // Invoke update when first called, debouncing subsequent calls.
+        var debounceUpdate = _.debounce(function() {
+          $scope.$evalAsync(update);
+        }, 250, {
+          leading: true,
+          trailing: false,
+          maxWait: 1000
+        });
+
+        // Set up the sort configuration for `pf-simple-sort`.
+        $scope.sortConfig = {
+          fields: [{
+            id: 'lastTimestamp',
+            title: 'Time',
+            sortType: 'alpha'
+          }, {
+            id: 'type',
+            title: 'Severity',
+            sortType: 'alpha'
+          }, {
+            id: 'reason',
+            title: 'Reason',
+            sortType: 'alpha'
+          }, {
+            id: 'message',
+            title: 'Message',
+            sortType: 'alpha'
+          }, {
+            id: 'count',
+            title: 'Count',
+            sortType: 'numeric'
+          }],
+          isAscending: false,
+          onSortChange: update
+        };
+
+        // Conditionally add kind and name to sort fields if not passed to the directive.
+        if (!$scope.resourceKind || !$scope.resourceName) {
+          $scope.sortConfig.fields.splice(1, 0, {
+            id: 'involvedObject.name',
+            title: 'Name',
+            sortType: 'alpha'
+          }, {
+            id: 'involvedObject.kind',
+            title: 'Kind',
+            sortType: 'alpha'
+          });
+        }
+
         var watches = [];
         watches.push(DataService.watch("events", $scope.projectContext, function(events) {
           $scope.emptyMessage = "No events to show";
           // $scope.eventsArray = $filter('toArray')(events.by("metadata.name"));
```

Comment:
Yeah, it wasn't needed since `_.filter` handles objects. Comment I made on another PR. I'll remove.